liberapay/models/participant.py

```diff
@@ -2176,8 +2182,49 @@ def update_avatar(self, src=None, cursor=None, avatar_email=None, check=True):
         if platform == 'libravatar' or platform is None and email:
             if not email:
                 return
-            avatar_id = md5(email.strip().lower().encode('utf8')).hexdigest()
-            avatar_url = 'https://seccdn.libravatar.org/avatar/'+avatar_id
+            # https://wiki.libravatar.org/api/
+            normalized_email = email.strip().lower()
+            try:
+                # Look up the SRV record to use.
+                _, email_domain = normalized_email.rsplit('@', 1)
+                try:
+                    srv_records = DNS.query('_avatars-sec._tcp.'+email_domain, 'SRV')
+                    scheme = 'https'
+                except Exception:
+                    srv_records = DNS.query('_avatars._tcp.'+email_domain, 'SRV')
+                    scheme = 'http'
+                # Filter down to just the records with the "highest" `.priority`
+                # (lower number = higher priority).
+                top_priority = min(rec.priority for rec in srv_records)
+                srv_records = [rec for rec in srv_records if rec.priority == top_priority]
+                # Of those, choose randomly based on their relative `.weight`s.
+                weight_choice = randint(0, sum(rec.weight for rec in srv_records))
+                weight_sum = 0
+                for rec in srv_records:
+                    weight_sum += rec.weight
+                    if weight_sum >= weight_choice:
+                        choice_record = rec
+                        break
```

Comment:
Is this algorithm standard? Can you add a link to where you found it?

Reply:
Yes, this is the standard RFC2782 algorithm (described at the top of page 4 of the RFC). I'll amend the commit to add a comment.

Also, I just noticed a bug in my implementation with zero-weight records.

Reply:
Done.